sycl/doc/extensions/SPIRV/SPV_INTEL_float_controls2.asciidoc

```diff
@@ -0,0 +1,206 @@
+
+
+Version
+-------
+
+[width="40%",cols="25,25"]
+|========================================
+| Last Modified Date | 2020-08-03
+| Revision           | 3
+|========================================
+
+Dependencies
+------------
+
+This extension is written against the SPIR-V Specification,
+Version 1.5, Revision 3, Unified
+
+This extension requires SPIR-V 1.0.
+
+Overview
+--------
+
+This extension adds new execution modes and decorations to control floating-point computations.
+This extension adds execution modes with round to positive infinity and round to negative infinity default rounding
+and execution modes specifying compliance with floating-point arithmetic (IEEE 754) standard.
+This extension adds new decorations that can be applied to a function to control floating-point computations inside the function.
+
+
+Extension Name
+--------------
+
+To use this extension within a SPIR-V module, the following
+*OpExtension* must be present in the module:
+
+----
+OpExtension "SPV_INTEL_float_controls2"
+----
+
+Modifications to the SPIR-V Specification, Version 1.5, Revision 3, Unified
+---------------------------------------------------------------------------
+
+Modify Section 3.6, Execution Mode, add the following rows to the Execution Mode table:
+
+--
+[cols="1,20,10,10",options="header",width = "80%"]
+|====
+  2+^| Execution Mode  | Extra Operands | Enabling Capabilities
+| 5620 | *RoundingModeRTPINTEL* +
+The default rounding mode for floating-point arithmetic and conversions instructions must be round to positive infinity.
+If an instruction is decorated with *FPRoundingMode* or defines a rounding mode in its description, that rounding mode is applied and *RoundingModeRTPINTEL* is ignored.
+Only affects instructions operating on a floating-point type whose component width is _Target Width_.
+
+_Target Width_ is an unsigned 32-bit integer.
+| <<Literal, 'Literal'>>  _Target Width_ | *RoundToInfinityINTEL*
+| 5621 | *RoundingModeRTNINTEL* +
+The default rounding mode for floating-point arithmetic and conversions instructions must be round to negative infinity.
+If an instruction is decorated with *FPRoundingMode* or defines a rounding mode in its description, that rounding mode is applied and *RoundingModeRTNINTEL* is ignored.
+Only affects instructions operating on a floating-point type whose component width is _Target Width_.
+
+_Target Width_ is an unsigned 32-bit integer.
+| <<Literal, 'Literal'>> _Target Width_ | *RoundToInfinityINTEL*
+  | 5622 | *FloatingPointModeALTINTEL* +
+The default floating-point operation mode for floating-point arithmetic and conversions instructions must be the Alternative Floating-Point Mode as specified by the client API.
+Only affects instructions operating on a floating-point type whose component width is _Target Width_.
+
+_Target Width_ is an unsigned 32-bit integer.
+| <<Literal, 'Literal'>> _Target Width_ | *FloatingPointModeINTEL*
+| 5623 | *FloatingPointModeIEEEINTEL* +
```

Comment:
Having both: execution mode `FloatingPointModeIEEEINTEL` and decoration `FunctionFloatingPointModeINTEL` seems redundant.  Could you clarify what is the difference between them?

And I have the same question for the `RoundingMode*INTEL` execution mode.

Reply:
The intention is that the execution modes set the default rounding/denorm/floating-point modes for the entire entry point. 
Decorations allow to overwrite the rounding/denorm/floating-point mode for all arithmetic and conversion instructions in a single function. 
I agree that the same result could be achieved with the decoration alone but could potentially require decorating each function in a module.

Reply:
I added the Issue 1) with the explanation.